### DISCUSSION TITLE
Automatic scoping for resource routes

### DISCUSSION THREAD

**Discussion by simonhir** [laravel/framework#33436] [General] [answered]
Hello,

is there a way to use the new automatic scoping in resource routes?

The following isn't working:
```
Route::apiResource('instances/{instance:id}/projects', 'ProjectController', [
    'except' => ['destroy']
]);
```

If isn't wouldn't it be a nice feature?

**Comment by jmarcher** [comment]
You need to use the API like this: 

```php
Route::apiResource('instances/{instance:id}/projects', 'ProjectController')->except('destroy');
```

**Reply by simonhir** [reply]
Thank you for the hint but doesn't change anything for the auto scoping even if I remove the except part complete.

**Reply by jmarcher** [reply]
What do you mean by "auto scoping"?

**Reply by simonhir** [reply]
I want to use the `Route Model Binding Improvements > Automatic Scoping` from [Laravel 7](https://laravel.com/docs/7.x/releases#laravel-7) for the resource route

**Reply by simonhir** [reply]
@jmarcher do you have any idea how to do that or if it is even possible?

**Comment by simonhir** [comment, marked as answer]
To use the auto scoping feature you need to replace the resource route parameter.
```
Route::apiResource('instances/{instance:id}/projects', 'ProjectController')->parameter('projects', 'project:id');
```
